lib/https.js

```diff
@@ -40,6 +40,12 @@ function Server(opts, requestListener) {
   }
   opts = util._extend({}, opts);
 
+  if (opts && typeof opts.key === 'boolean')
```

Comment:
I don't think checking against all invalid values is the way to go. I think it should confirm it is a valid value and reject everything else.